flake8_pyi/visitor.py

```diff
@@ -442,17 +436,12 @@ def _is_bad_TypedDict(node: ast.Call) -> bool:
     if not isinstance(typed_dict_annotations, ast.Dict):
         return False
 
-    typed_dict_fields = typed_dict_annotations.keys
-
-    if not _is_list_of_str_nodes(typed_dict_fields):
-        return False
-
-    fieldnames = [field.value for field in typed_dict_fields]
-
-    return all(
-        fieldname.isidentifier() and not iskeyword(fieldname)
-        for fieldname in fieldnames
-    )
+    for key in typed_dict_annotations.keys:
+        if not isinstance(key, ast.Constant) or type(key.value) is not str:
+            return False
+        if not key.value.isidentifier() or iskeyword(key.value):
+            return False
```

Comment:
nit: I'd probably either do

```suggestion
        if not isinstance(key, ast.Constant):
            return False
        if type(key.value) is not str:
            return False
        if not key.value.isidentifier():
            return False
        if iskeyword(key.value):
            return False
```

or

```suggestion
        if not (
            isinstance(key, ast.Constant)
            and type(key.value) is str
            and key.value.isidentifier()
            and not iskeyword(key.value)
        ):
            return False
```